controllers/idler/idler_controller.go

```diff
@@ -141,6 +153,90 @@ func (r *Reconciler) ensureIdling(logger logr.Logger, idler *toolchainv1alpha1.I
+	_, found := condition.FindConditionByType(idler.Status.Conditions, toolchainv1alpha1.IdlerTriggeredNotificationCreated)
+	if !found || condition.IsFalse(idler.Status.Conditions, toolchainv1alpha1.IdlerTriggeredNotificationCreated) {
+		notificationName := fmt.Sprintf("%s-%s", idler.Name, toolchainv1alpha1.NotificationTypeIdled)
+		notification := &toolchainv1alpha1.Notification{}
+		// Check if notification already exists in host
+		if err := hostCluster.Client.Get(context.TODO(), types.NamespacedName{Name: notificationName, Namespace: hostCluster.OperatorNamespace}, notification); err != nil {
+			if errors.IsNotFound(err) {
+				userEmails, err := r.getUserEmailsFromMURs(logger, hostCluster, idler)
+				if err != nil {
+					return err
+				}
+				if len(userEmails) > 0 {
+					keysAndVals := map[string]string{
+						"Namespace": idler.Name,
+					}
+					for _, userEmail := range userEmails {
+						_, err := notify.NewNotificationBuilder(hostCluster.Client, hostCluster.OperatorNamespace).
+							WithName(notificationName).
+							WithNotificationType(toolchainv1alpha1.NotificationTypeIdled).
+							WithTemplate("idlertriggered").
+							WithKeysAndValues(keysAndVals).
+							Create(userEmail)
+						if err != nil {
+							return errs.Wrapf(err, "Unable to create Notification CR from Idler")
+						}
+					}
+				} else {
+					// no email found, thus no email sent
+					logger.Error(fmt.Errorf("no email found for the user in MURs"), "no email found to ")
+					return nil
+				}
+			} else {
+				return err
+			}
+		}
+		// set notification created condition
+		if err := r.setStatusIdlerNotificationCreated(idler); err != nil {
+			return err
+		}
+		return nil
+	}
+	//notification already created
+	return nil
+}
+
+func (r *Reconciler) getUserEmailsFromMURs(logger logr.Logger, hostCluster *cluster.CachedToolchainCluster, idler *toolchainv1alpha1.Idler) ([]string, error) {
+	logger.Info("getUserEmailsFromMURs")
+	var emails []string
+	//get NSTemplateSet from idler
+	if owner, found := idler.GetLabels()[toolchainv1alpha1.OwnerLabelKey]; found {
+		nsTemplateSet := &toolchainv1alpha1.NSTemplateSet{}
+		err := r.Client.Get(context.TODO(), types.NamespacedName{Name: owner, Namespace: r.Namespace}, nsTemplateSet)
+		if err != nil {
+			logger.Error(err, "Could not get the NSTemplateSet with name", "owner", owner)
+			return emails, err
+		}
+		// iterate on space roles from NSTemplateSet
+		var murs []string
+		for _, spaceRole := range nsTemplateSet.Spec.SpaceRoles {
+			murs = append(murs, spaceRole.Usernames...)
+		}
+		// get MUR from host and use user email from annotations
+		for _, mur := range murs {
+			getMUR := &toolchainv1alpha1.MasterUserRecord{}
+			err := hostCluster.Client.Get(context.TODO(), types.NamespacedName{Name: mur, Namespace: hostCluster.OperatorNamespace}, getMUR)
+			if err != nil {
+				return emails, fmt.Errorf("Could not get the MUR with name: %s ", mur)
```

Comment:
By returning a new error here we're losing the error that came from the client so either we should just return the error from the client or wrap the error with our own error msg. Also, if we're wrapping the error with another message then same as previous comment it should start with a lowercase letter.

Reply:
wrapped the error and changed to lowercase in [67a0dde](https://github.com/codeready-toolchain/member-operator/pull/357/commits/67a0dde897ead65c14667363a42ff694ae962c68)